okhttp/src/jvmMain/kotlin/okhttp3/ConnectionSpec.kt

```diff
@@ -67,6 +76,15 @@ class ConnectionSpec internal constructor(
       level = DeprecationLevel.ERROR)
   fun cipherSuites(): List<CipherSuite>? = cipherSuites
 
+  /**
+   * Returns the TLS versions to use when negotiating a connection. Returns null if all of the SSL
+   * socket's enabled TLS versions should be used.
+   */
+  @get:JvmName("tlsVersionsAsString") val tlsVersionsAsStringList: List<String>?
+    get() {
+      return tlsVersionsAsString?.let { Collections.unmodifiableList(it.asList()) }
```

Comment:
I'm not sure the intent is to support arbitrary TlsVersion here. Let me check when we have previously discussed.

But I'm not necessarily against it.

Like with cipherSuite, is we do want to support, we should probably make the types open.

Reply:
With this change, anyone using the current tlsVersions will get a nasty surprise.

`tlsSpec.tlsVersions()`

failing with

```
java.lang.IllegalArgumentException: Unexpected TLS version: TLS9k
	at okhttp3.TlsVersion$Companion.forJavaName(TlsVersion.kt:47)
	at okhttp3.ConnectionSpec.tlsVersions(ConnectionSpec.kt:85)
```